src/reflect/scala/reflect/internal/Definitions.scala

```diff
@@ -712,6 +712,7 @@ trait Definitions extends api.StandardDefinitions {
     def tupleComponents(tp: Type)      = tp.dealiasWiden.typeArgs
 
     lazy val ProductRootClass: ClassSymbol = requiredClass[scala.Product]
+    lazy val ProductRootTpe: Type          = typeOf[scala.Product]
```

Comment:
`ProductRootClass.tpe` seems to be more in line with what is done elsewhere in this file.